filler.go

```diff
@@ -92,8 +91,17 @@ func (r *routeFiller) fill(ctx reflect.Value) error {
 	}
 	if r.matchedPathIndex != -1 {
 		i := r.routeParams[r.matchedPathIndex]
+		paths := r.reqURL[r.matchedPathIndex:]
+		path := bytes.NewBuffer([]byte(``))
+		for _, v := range paths {
+			path.WriteString(v)
+			path.WriteRune('/')
+		}
```

Comment:
Can you try [String.Builder](https://golang.org/pkg/strings/#Builder) ?
But this requires the minimax version is 1.10.
or you want to try [copy](https://stackoverflow.com/questions/1760757/how-to-efficiently-concatenate-strings-in-go/23857998#23857998) :D

Reply:
`strings.Builder` do not bring a big difference here, I'm going to try `copy` XD

Reply:
`copy` can bring 10% improvement for `BenchmarkRequest/WithUserDefinedContext-4`

original:
```
BenchmarkRequest/WithoutUserDefinedContext-4         	 1000000	      1375 ns/op	     928 B/op	      11 allocs/op
BenchmarkRequest/WithUserDefinedContext-4            	  500000	      2233 ns/op	    1280 B/op	      21 allocs/op
BenchmarkRequest/WithCustomResponse-4                	 1000000	      2004 ns/op	    1336 B/op	      14 allocs/op
BenchmarkRequest/WithHeader-4                        	 1000000	      2578 ns/op	    1216 B/op	      19 allocs/op
```

`copy` version:
```
BenchmarkRequest/WithoutUserDefinedContext-4         	 1000000	      1193 ns/op	     928 B/op	      11 allocs/op
BenchmarkRequest/WithUserDefinedContext-4            	 1000000	      2011 ns/op	    1280 B/op	      21 allocs/op
BenchmarkRequest/WithCustomResponse-4                	 1000000	      1683 ns/op	    1336 B/op	      14 allocs/op
BenchmarkRequest/WithHeader-4                        	 1000000	      1981 ns/op	    1216 B/op	      19 allocs/op
```

Reply:
have a try with `filepath.Join` and it actually close with `copy` version and no need to maintain it, I would prefer filepath solution, what do you think?

Reply:
Oops, its effect performance a lot, seems it's not stable

```
BenchmarkRequest/WithoutUserDefinedContext-4         	 1000000	      1539 ns/op	     928 B/op	      11 allocs/op
BenchmarkRequest/WithUserDefinedContext-4            	  500000	      2433 ns/op	    1280 B/op	      21 allocs/op
BenchmarkRequest/WithCustomResponse-4                	 1000000	      2271 ns/op	    1336 B/op	      14 allocs/op
BenchmarkRequest/WithHeader-4                        	  500000	      3042 ns/op	    1216 B/op	      19 allocs/op
```

Reply:
IIRC, we have another place are using the same code, so please change that too.
and wrap this to a function and give a proper name, thanks.

Reply:
Could you find out the usage for me? And I would put it at `internal/filepath`, named `Join`

Reply:
@dannypsnl  	https://github.com/dannypsnl/rocket/blob/bc498f7c41f5479bd481db9f94ea34b4dad63374/handler.go#L94

Reply:
I think it's outdated, I couldn't find any usage now.